### PR TITLE
update typesafe.artifactory-online.com to private-repo

### PR DESCRIPTION
**Pull request by adriaanm**
backport of #2880

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -364,9 +364,9 @@ TODO:
     <property name="sbt.lib.dir"           value="${build-sbt.dir}/${sbt.latest.version}/lib"/>
 
     <property name="sbt.interface.jar"     value="${sbt.lib.dir}/interface.jar"/>
-    <property name="sbt.interface.url"     value="http://typesafe.artifactoryonline.com/typesafe/ivy-releases/org.scala-sbt/interface/${sbt.latest.version}/jars/interface.jar"/>
+    <property name="sbt.interface.url"     value="http://private-repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/interface/${sbt.latest.version}/jars/interface.jar"/>
     <property name="sbt.interface.src.jar" value="${sbt.src.dir}/compiler-interface-src.jar"/>
-    <property name="sbt.interface.src.url" value="http://typesafe.artifactoryonline.com/typesafe/ivy-releases/org.scala-sbt/compiler-interface/${sbt.latest.version}/jars/compiler-interface-src.jar"/>
+    <property name="sbt.interface.src.url" value="http://private-repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compiler-interface/${sbt.latest.version}/jars/compiler-interface-src.jar"/>
```
Comment (retronym) [inline build.xml:369]: Is the old URL going away? Will that mean we can't build old SHAs anymore?

Comment (adriaanm) [inline build.xml:369]: oh shit! good point.... To be honest, I don't know what's going on. Just the rebaser for @huitseeker's fixes of the changes made by @jsuereth

Comment (retronym) [inline build.xml:369]: Worst case, we could perform the historical builds with: `ant -Dsbt.interface.url=...  build`.

Comment (jsuereth) [inline build.xml:369]: the old repo is fine.  HTTP:// works fine. it's https:// (publishing) that's borked due to a cert failure in JDK7.

**Comment by jsuereth** [comment]
This LGTM

**Comment by huitseeker** [comment]
This LGTM

**Comment by gkossakowski** [comment]
Thanks.

**Pull request merged**